### DISCUSSION TITLE
Absence of video download progress indicator/interface

### DISCUSSION THREAD

**Discussion by JulesFM92** [FreeTubeApp/FreeTube#4191] [General]
Hello there.

First of all, I'm just a common user, not a programmer by any means, and I would like to start this topic just to point out something that I have found to be missing just today. For the first time, I decided to download a video using the integrated button right on the player interface, and then I saw that, after choosing the directory where to save the video file, only a small text appeared on the corner of the screen, saying that the download had started. It vanished shortly, and I have not found anything, anywhere on the FreeTube interface, that could give me a visual indication of the download progress, estimated time, or even if it was or not successful.

I imagine that, once the download is complete, a small text will tell me that, and maybe I have not looked well enough, but, as an end user, I must point out that the lack of a visual, easily accessible interface element with detailed information about an ongoing download is a problem, because we become effectively blind and ignorant, and with no means even to make certain decisions. The specific video I am downloading right now (or that I must trust that is being downloaded) is big, so I expect that it will take a long time, however, if there were, say, a downloads tab, or equivalent visual element that the user could open anytime, showing the progress, expected time of transfer, and maybe a cancel command, the user could know in advance how much to wait, or decide that it would be better to do the download in a different moment. As it is, the user has no choice but to wait blindly, which leaves us insecure.

I know that this is a voluntary project, I love your app and I am very grateful for all your great work. Just wanted to make this contribution. I don't know if this would be categorized as a feature request, but I hope this is something possible and simple to implement.

Best regards.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @JulesFM92 

You are right users dont have a way to check what the progress is. The only thing the user get to know if when a download starts, finished and if there went something wrong with the download.

This was reported before in https://github.com/FreeTubeApp/FreeTube/issues/2110 but was closed because this isnt a bug in the sense of that this was the intended behavior of the PR that created this feature.

Download progress is something that will be included in https://github.com/FreeTubeApp/FreeTube/issues/1772#issuecomment-1689720206 but the Download Tab isn't a high priority because there are allot of other things on the list that needs our attention.

So it can take a while before we get to it but we are happy to accept anyone's PR from outside our team that would like to contribute to a Download Tab.

**Reply by JulesFM92** [reply]
Thank you very much for the reply, I understand completely. Sorry if I bothered.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
You're welcome, dont worry about it u didn't bother me at all. I'm happy that i could help you understand :)
